### PR TITLE
Support the dns_powerdns plugin

### PR DESCRIPTION
**Pull request by ekohl**
This is untested and just something to refer to for now.

### CODE REVIEW

**Comment by ekohl** [comment]
I'm considering adding flags to manage mysql or at least the grant. Thoughts?

**Comment by ekohl** [comment]
I split it into a separate commit for easier review if it's desired.

**Comment by domcleal** [comment]
Managing the DB sounds like a useful feature, at least until the plugin has API support.

**Comment by mmoll** [comment]
This is looking good and the mysql module is already in the installer module set anyway.

**Comment by domcleal** [comment]
Perhaps add a little note to the README to indicate the soft dependency from this class?

**Comment by ekohl** [comment]
Something like this?

**Comment by domcleal** [comment]
Yep.  Looks fine to me.  Do you want to test compilation with the DB management?

**Comment by ekohl** [comment]
Sounds like a good idea. I also plan to test this module on a real system and then remove ```[WIP]``` somewhere this week.

File: README.md [hunk 1]
```diff
@@ -5,6 +5,21 @@ interact with, e.g. DNS, DHCP and Puppet and TFTP.
...
+## PowerDNS support
+
+To use the PowerDNS plugin, the following variables need to be set on the main
+`foreman_proxy` class.
+
+    $dns          => true
+    $dns_managed  => false
+    $dns_provider => 'dns_powerdns'
+
+Then you also need to include `foreman::plugin::dns::powerdns`.
```
Comment (ekohl) [inline README.md]: s/foreman/foreman_proxy/

**Comment by ekohl** [comment]
Will rebase when #190 is merged because it doesn't work without it.

**Comment by ekohl** [comment]
Verified it works on Jessie.

**Comment by mmoll** [comment]
:+1:

**Comment by domcleal** [comment]
Looks good to me, but should we also test the class with `manage_database => true`?

**Comment by ekohl** [comment]
Will do

File: .sync.yml [hunk 2]
```diff
@@ -3,3 +3,4 @@ Rakefile:
   param_docs_pattern:
     - manifests/init.pp
     - manifests/plugin/*.pp
+    - manifests/plugin/dns/*.pp
```
Comment (domcleal) [inline .sync.yml]: This'll need a rebase after 39af736, but the fix in that commit will probably suffice for you too.

**Comment by ekohl** [comment]
Updated

**Comment by domcleal** [comment]
:+1:, though #196 should fix the unrelated test failures.

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

**Pull request closed without merging**